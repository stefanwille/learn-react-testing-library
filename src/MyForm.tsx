import React, { useState } from "react";
import { Formik, Form, FormikProps, FormikActions } from "formik";

interface FormState {
  firstName: string;
  lastName: string;
}

export const MyForm = () => {
  const [fullName, setFullName] = useState("");
  return (
    <div>
      MyForm
      <Formik
        initialValues={{ firstName: "", lastName: "" }}
        onSubmit={(values: FormState, actions: FormikActions<FormState>) => {
          const { firstName, lastName } = values;
          setFullName(`${firstName} ${lastName}`);
        }}
        render={(formikApi: FormikProps<FormState>) => {
          return (
            <Form autoComplete="off">
              <p>Hello</p>
              <input
                autoComplete="off"
                type="text"
                name="firstName"
                value={formikApi.values.firstName}
                onChange={formikApi.handleChange}
              />
              <br />
              <input
                autoComplete="off"
                type="text"
                name="lastName"
                value={formikApi.values.lastName}
                onChange={formikApi.handleChange}
              />
              <input type="submit" value="Submit"></input>
            </Form>
          );
        }}
      ></Formik>
      {fullName.length > 0 && <div>Your name is {fullName}</div>}
    </div>
  );
};
