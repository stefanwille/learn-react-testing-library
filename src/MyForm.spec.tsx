import React from "react";

import { render, fireEvent, wait } from "@testing-library/react";
import { MyForm } from "./MyForm";

describe("MyForm", () => {
  it("renders", () => {
    const container = render(<MyForm />);

    const firstName = container.getByLabelText("First name:");
    expect(firstName).not.toBeNull();

    const lastName = container.getByLabelText("Last name:");
    expect(lastName).not.toBeNull();

    const submit = container.getByDisplayValue("Submit");
    expect(submit).not.toBeNull();
  });

  it("handles submit", async () => {
    const container = render(<MyForm />);

    const firstName = container.getByLabelText("First name:");

    const lastName = container.getByLabelText("Last name:");

    const submit = container.getByDisplayValue("Submit");

    fireEvent.change(lastName, { target: { value: "Slobodan" } });
    fireEvent.change(firstName, { target: { value: "Stefan" } });
    fireEvent.click(submit);
    await wait(() =>
      expect(container.getByTestId("form-result").textContent).toBe(
        "Your name is Stefan Slobodan"
      )
    );

    // await waitForElement(() => container.getByTestId("form-result"));
    // expect(container.getByTestId("form-result").textContent).toBe(
    //   "Your name is Stefan Slobodan"
    // );
    // const fullName = await container.findByTestId("form-result");
    // expect(fullName.textContent).toMatch(/Your name is Stefan Slobodan/);
  });
});
