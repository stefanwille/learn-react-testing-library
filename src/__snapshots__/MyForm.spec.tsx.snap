// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyForm handles submit 1`] = `
<div
  data-testid="form-result"
>
  Your name is 
  Stefan Slobodan
</div>
`;
